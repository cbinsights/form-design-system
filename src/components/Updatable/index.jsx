import React from 'react';
import PropTypes from 'prop-types';

const Updatable = ({ message, children }) => (
  <div className="updatable">
    {children}
    {message && <div className="updatable-indicator">{message}</div>}
  </div>
);

Updatable.propTypes = {
  /** Count or message for the updates bubble */
  message: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
  /** React children */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
};

export default Updatable;
