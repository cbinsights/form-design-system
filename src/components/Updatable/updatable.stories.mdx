import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import { withKnobs, text } from '@storybook/addon-knobs';

import CheckIcon from 'lib/icons/react/CheckIcon';

import { ImportPath } from 'components/util/storybook';

import Updatable from './';

<Meta title="Components/Updatable" decorators={[withKnobs]} component={Updatable} />

# Updatable

<ImportPath component={Updatable} section="layout" />

Wraps children in an "updatable" container with a `message` displayed as a notification-style red bubble.

<Story name="Knobs" parameters={{ docs: { disable: true }}}>
  <Updatable message={text('message', 42)}>
    <button>lorem ipsum</button>
  </Updatable>
</Story>

<Preview isExpanded>
  <Updatable message="42">
    <button>lorem ipsum</button>
  </Updatable>
</Preview>

<Props of={Updatable} />

### Different Variations

<Preview>
  <Story name="Variations">
    <React.Fragment>
      <Updatable message="alert">
        <div
          style={{
            border: '1px dashed #f09',
            padding: '5px',
          }}
        >
          Updatable can wrap <b>anything</b>!
        </div>
      </Updatable>
      <br />
      <br />
      <Updatable message={<CheckIcon size="xs" color="#fff" />}>
        <img src="https://placeimg.com/80/81/animals" />
      </Updatable>
    </React.Fragment>
  </Story>
</Preview>