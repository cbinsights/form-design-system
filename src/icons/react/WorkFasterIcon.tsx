/**
  * DO NOT EDIT',
  * Generated by fds-dictionary on Wed May 04 2022 16:41:34 GMT-0400 (Eastern Daylight Time)
  * github.com/cbinsights/form-design-system/
  */

import React from 'react';
import { IconProps } from 'components/Icon';

const WorkFasterIcon = ({
  size = 's',
  color,
  customSize,
}: IconProps): JSX.Element => (
  <div
    className={`fds-icon fds-icon--${size}`}
    style={{
      fill: color,
      width: customSize && `${customSize}px`,
      height: customSize && `${customSize}px`,
    }}
  >
    <svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
	<g fillRule="evenodd">
		<g>
			<polygon points="13.0233982 10 18 10 10.9891409 23 10.9891409 14 6 14 13.0233982 1"/>
		</g>
	</g>
</svg>
  </div>
);

export default WorkFasterIcon;

