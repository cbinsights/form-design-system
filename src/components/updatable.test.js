import React from 'react';
import { shallow } from 'enzyme';

import WorkFasterIcon from '../../lib/icons/react/WorkFasterIcon';
import Updatable from './Updatable';

const MOCK_COMPONENT = (
  <div>I can be updated with an update indicator.</div>
);

describe('AvatarRow component', () => {

  it('matches snapshot with count', () => {
    const component = shallow(<Updatable message="666">{MOCK_COMPONENT}</Updatable>)
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot with element in message', () => {
    const message = (<span><WorkFasterIcon size="xs" color="#ffffff" /> lightning bolt for some reason</span>);
    const component = shallow(<Updatable message={message}>{MOCK_COMPONENT}</Updatable>);
    expect(component).toMatchSnapshot();
  });

});
