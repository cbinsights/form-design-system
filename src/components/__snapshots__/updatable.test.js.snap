// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarRow component matches snapshot with count 1`] = `
<div
  className="updatable"
>
  <div>
    I can be updated with an update indicator.
  </div>
  <div
    className="updatable-indicator"
  >
    666
  </div>
</div>
`;

exports[`AvatarRow component matches snapshot with element in message 1`] = `
<div
  className="updatable"
>
  <div>
    I can be updated with an update indicator.
  </div>
  <div
    className="updatable-indicator"
  >
    <span>
      <WorkFasterIcon
        color="#ffffff"
        size="xs"
      />
       lightning bolt for some reason
    </span>
  </div>
</div>
`;
